File: languages/go/entity/identity/datetime_attribute_descriptor.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// DatetimeAttributeDescriptor represents a descriptor for an attribute with a datetime value.
type DatetimeAttributeDescriptor struct {
	AttributeDescriptor
}

// NewDatetimeAttributeDescriptor creates a new DatetimeAttributeDescriptor instance with the provided details.
func NewDatetimeAttributeDescriptor(name string, id string, description string, required bool) DatetimeAttributeDescriptor {
	a := AttributeDescriptor{
		DisplayName: name,
		Id:          id,
		Description: description,
		Required:    required,
	}
	return DatetimeAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func NewDatetimeAttributeDescriptorFromProto(s *proto.DateTimeAttributeDefinition) DatetimeAttributeDescriptor {
	if s == nil {
		return DatetimeAttributeDescriptor{}
	}
	a := AttributeDescriptor{
		DisplayName: s.GetId(),
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
	return DatetimeAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func (s DatetimeAttributeDescriptor) ToProto() *proto.DateTimeAttributeDefinition {
	return &proto.DateTimeAttributeDefinition{
		DisplayName: s.DisplayName,
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
}

File: languages/js/src/entity/identity/datetime_attribute_descriptor.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities";
import { AttributeDescriptor } from "./attribute_descriptor";

/**
 * Represents a descriptor for an attribute with a datetime value.
 */
export class DateTimeAttributeDescriptor extends AttributeDescriptor {
  public toProto(): identityEntitiesProto.DateTimeAttributeDefinition {
    return identityEntitiesProto.DateTimeAttributeDefinition.fromPartial({
      displayName: this.displayName,
      id: this.technicalName,
      description: this.description,
      required: this.required
    });
  }

  static fromProto(
    r: identityEntitiesProto.DateTimeAttributeDefinition
  ): DateTimeAttributeDescriptor {
    return new DateTimeAttributeDescriptor(
      r.displayName,
      r.id,
      r.description,
      r.required
    );
  }
}
